### PR TITLE
Issue #2615 Disable special Char parsing for proxy for Env variable

### LINKED ISSUE #2615: Do not escape special charecters from proxy system environment variables
Escaping proxy username/password using special characters is not a minishift's job for system set environment variables.
The proxy env variables (http_proxy and friends) must contain **already** escaped values.

### PR DESCRIPTION
**Pull request by praveenkumar**
Fixes #2615

### CODE REVIEW

**Comment by praveenkumar** [comment]
retest this please.

**Comment by LalatenduMohanty** [comment]
Have anyone tested the PR?

**Comment by amitkrout** [comment]
I will validate it from my end once

**Comment by amitkrout** [comment]
Pinged op https://github.com/minishift/minishift/issues/2615#issuecomment-412413291 to test it out

**Comment by gbraad** [comment]
Any progress?

File: pkg/util/proxy.go [hunk 1]
```diff
@@ -41,6 +41,15 @@ type ProxyConfig struct {
 func NewProxyConfig(httpProxy string, httpsProxy string, noProxy string) (*ProxyConfig, error) {
 	defaultScheme := "http"
 
+	// We only parse the Proxy Special char if it is coming from the config file
+	// or user defined flag. In case of Env variable we just ignore parsing.
```
Comment (amitkrout) [inline pkg/util/proxy.go]: I think this need to be documented somewhere in https://docs.okd.io/latest/minishift/using/basic-usage.html#http-s-proxies

File: docs/source/using/basic-usage.adoc [hunk 2]
```diff
@@ -300,9 +300,7 @@ The environment variables can be specified in lower case or upper case.
 The lower case variables take higher precedence over upper case variables.
 If these variables are set, they are implicitly used during `minishift start` unless explicitly overridden by the corresponding command line flags.
 
-- Using the proxy options requires that you run OpenShift version 1.5.0 or later.
-Use the `openshift-version` option to request a specific version of OpenShift.
-You can list all {project}-compatible OpenShift versions with the xref:../command-ref/minishift_openshift_version_list.adoc#[`minishift openshift version list`] command.
+- The environment variables are not parsed to check special character in proxy so user should replace any special character manually before setting it.
```
Comment (LalatenduMohanty) [inline docs/source/using/basic-usage.adoc]: @praveenkumar This is confusing as it is not clear why user need to replace special character?

Comment (LalatenduMohanty) [reply docs/source/using/basic-usage.adoc]: @praveenkumar  suggestion `Minishift does not escape the special characters in the environment variables for proxy. So user need to escape special characters for Minishift to get the correct values.`

**Pull request merged**